src/pages/product/ProductPage.tsx

```diff
@@ -1,49 +1,52 @@
-import React from 'react'
-import { Box, Composition } from 'atomic-layout'
-import { ProductItem } from '../../molecules/ProductItem'
+import React, { useState, useEffect } from 'react'
+import { RouteComponentProps } from 'react-router'
+import { Box } from 'atomic-layout'
+import { RecomendedProducts } from '../../molecules/RecomendedProducts'
 import { Tabs } from '../../molecules/Tabs'
 import { ProductSummary } from '../components/ProductSummary'
 
-const data = [
-  {
-    image:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/1-21-620x698.jpg',
-    secondImage:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/2-21-600x675.jpg',
-    name: 'Mega 2 Seater Sofa',
-    price: 160.71,
-    buttonText: 'add to card'
-  },
-  {
-    image:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/1-28-600x675.jpg',
-    secondImage:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/2-28-600x675.jpg',
-    name: 'Sentei Pruning Shears',
-    price: 141.36,
-    buttonText: 'add to card'
-  },
-  {
-    image:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/1-21-620x698.jpg',
-    secondImage:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/2-21-600x675.jpg',
-    name: 'Mega 2 Seater Sofa',
-    price: 160.71,
-    buttonText: 'add to card'
-  },
-  {
-    image:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/1-28-600x675.jpg',
-    secondImage:
-      'https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/2-28-600x675.jpg',
-    name: 'Sentei Pruning Shears',
-    price: 141.36,
-    buttonText: 'add to card'
+export interface ProductData {
+  id: string
+  title: string
+  description: string
+  rating: number
+  price: number
+  images: string[]
+  relatedProducts: string[]
+  shopAttributes: ProductAttributes[]
+  reviews: ProductReview[]
+}
+
+export interface ProductAttributes {
+  title: string
+  data: number | string
+}
+
+export interface ProductReview {
+  name: string
+  date: string
+  rating: number
+  text: string
+}
+
+const ProductPage: React.FC<RouteComponentProps<{
+  productId: string
+}>> = ({ match }) => {
+  const [productData, setProductData] = useState<ProductData>()
+  const productId = match.params.productId
+
+  useEffect(() => {
+    fetch(
```

Comment:
If you be interested, we could isolate this fetch + useState logic into our _own custom React hook_. Boy, that'd be fun.

Reply:
Yes! Let's do it.